src/zenml/zen_stores/migrations/versions/722392c91006_make_is_service_account_mandatory.py

```diff
@@ -0,0 +1,86 @@
+"""make is_service_account mandatory [722392c91006].
+
+Revision ID: 722392c91006
+Revises: 0.46.1
+Create Date: 2023-11-13 15:38:26.253610
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.sql import text
+
+# revision identifiers, used by Alembic.
+revision = "722392c91006"
+down_revision = "0.46.1"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    connection = op.get_bind()
+
+    # Get all users where `is_service_account` is null
+    get_null_is_service_account = text(
+        """
+        SELECT id, name
+        FROM user
+        WHERE is_service_account IS NULL
+        """
+    )
+    result = connection.execute(get_null_is_service_account)
+
+    # For each user, if another user with the same name exists and has
+    # `is_service_account` set to false, delete it
+    for row in result:
+        id, name = row
+        get_other_user = text(
+            """
+            SELECT id
+            FROM user
+            WHERE name = :name
+            AND id != :id
+            AND is_service_account = 0
+            """
+        )
+        other_user_result = connection.execute(
+            get_other_user, {"name": name, "id": id}
+        )
+        for other_user_row in other_user_result:
+            other_user_id = other_user_row[0]
+            delete_other_user = text(
+                """
+                DELETE FROM user
+                WHERE id = :id
+                """
+            )
+            connection.execute(delete_other_user, {"id": other_user_id})
```

Comment:
This will not work,. if upgrade/downgrade was attempt multiple time, since only one user will have 0 others will have NULL

Reply:
![image](https://github.com/zenml-io/zenml/assets/31008759/fc0cec60-db78-4533-93c5-9391b37cc874)

Reply:
This is still not working for the case with multiple upgrade attempts:
![image](https://github.com/zenml-io/zenml/assets/31008759/268a04fd-25e1-4e7b-81c1-3ecee2d38346)

Reply:
Can we just take the earliest created user with default name and rename the rest?

Reply:
@avishniakov as Alexej pointed out, downgrades are not officially supported. Aside from you being able to do this locally, there is nothing in the documentation even remotely mentioning alembic, let alone how to configure it and use it to perform DB downgrades. The other reason we're not putting too much emphasis on this upgrade/downgrade/upgrade issue is that we yanked the release in time and there are very few users affected, if any.